genie-web/src/main/java/com/netflix/genie/web/services/impl/JobResolverServiceImpl.java

```diff
@@ -672,4 +724,214 @@ private int resolveJobMemory(final JobRequest jobRequest, final Command command)
+            } else if (commands.size() == 1) {
+                command = commands
+                    .stream()
+                    .findFirst()
+                    .orElseThrow(() -> new GenieJobResolutionException("No command matching criterion found."));
+                log.debug("Found single command {} matching criterion {}", command.getId(), criterion);
+            } else {
+                try {
+                    final ResourceSelectionResult<Command> result
+                        = this.commandSelector.selectCommand(commands, jobRequest);
+                    command = result
+                        .getSelectedResource()
+                        .orElseThrow(
+                            () -> new GenieJobResolutionException(
+                                "Expected a command but "
+                                    + result.getSelectorClass().getSimpleName()
+                                    + " didn't select anything. Rationale: "
+                                    + result.getSelectionRationale().orElse(NO_RATIONALE)
+                            )
+                        );
+                    log.debug(
+                        "Selected command {} for criterion {} using {} due to {}",
+                        command.getId(),
+                        criterion,
+                        result.getSelectorClass().getName(),
+                        result.getSelectionRationale().orElse(NO_RATIONALE)
+                    );
+                } catch (final ResourceSelectionException selectionException) {
+                    // TODO: Improve error handling?
+                    throw new GenieJobResolutionException(selectionException);
+                }
+            }
+
+            MetricsUtils.addSuccessTags(tags);
+            tags.add(Tag.of(MetricsConstants.TagKeys.COMMAND_ID, command.getId()));
+            tags.add(Tag.of(MetricsConstants.TagKeys.COMMAND_NAME, command.getMetadata().getName()));
+            return command;
+        } catch (final Throwable t) {
+            MetricsUtils.addFailureTagsWithException(tags, t);
+            if (t instanceof GenieJobResolutionException) {
+                throw t;
+            } else {
+                throw new GenieJobResolutionException(t);
+            }
+        } finally {
+            this.registry
+                .timer(RESOLVE_COMMAND_TIMER_NAME, tags)
+                .record(System.nanoTime() - start, TimeUnit.NANOSECONDS);
+        }
+    }
+
+    private Cluster resolveCluster(
+        final Command command,
+        final JobRequest jobRequest,
+        final String jobId
+    ) throws GenieJobResolutionException {
+        final long start = System.nanoTime();
+        final Set<Tag> tags = Sets.newHashSet();
+        int queryCount = 0;
+        int criteriaCombinationCount = 0;
+        try {
+            Cluster cluster = null;
+
+            // This makes it so that if the command has no Cluster Criteria no cluster will possibly
```

Comment:
I think the current behavior you describe is the right one.
If a command wants to match everything and just delegate to job criteria, it can always do so by setting a cluster criteria with `status=UP`.